### PR TITLE
A Lenksjö Happy Thoughts API

### PR DESCRIPTION
**Pull request by annalenksjo**
I managed to get the backend and frontend connected on the local level but then I had some issues with deploying my frontend to netlify. This was a problem I had previously as well with the project because I changed a folder-name.. I have a 1-1 booked so I hope I will solved it by the end of next week :-) 

Otherwise I am really happy with the project, and proud to have a fullstack application!

### CODE REVIEW

**Review by Yennie-Leek — APPROVED** [review]
Great job on doing a fullstack project!! 🙌 🥳 🙌
The code looks nice, clean and understandable. 
The frontend also looks good, but I understand if you want to add the small special features too to make it that little extra!😀
Well done!! 👍

File: server.js [hunk 1]
```diff
@@ -1,30 +1,115 @@
...
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Message is required"], 
+    trim: true,
+    minlength: 5,
+    maxlength: 140,
```
Comment (KarolCislowski) [inline server.js:20]: Nicely configured field 🤓

File: server.js [hunk 2]
```diff
@@ -1,30 +1,115 @@
...
+// ROUTES
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Get
+app.get('/thoughts', async (req, res) => {
+  try {
+    const allThoughts = await Thought.find().sort({ createdAt: -1 }).limit(20)
```
Comment (KarolCislowski) [inline server.js:48]: 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,115 @@
...
+  } catch (error) {
+    res.status(400).json(error)
+  }
+})
+
+app.post('/thoughts/:id/likes', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const updatedThought = await Thought.findByIdAndUpdate( { _id: id }, { $inc: { hearts: 1 }})
```
Comment (KarolCislowski) [inline server.js:68]: 💯

File: server.js [hunk 4]
```diff
@@ -1,30 +1,115 @@
...
+    const updatedThought = await Thought.findByIdAndUpdate(id, { message: req.body.message }, { new: true } )
+    
+    if (updatedThought) {
+      res.json(updatedThought) 
+    } else {
+      res.status(404).json({ message: 'Not found' })
+    }
+  } catch (error) { 
+    res.status(400).json({ message: 'invalid request', error })
+  }
```
Comment (KarolCislowski) [inline server.js:108]: Remember to restrict access to these endpoints because not all users should have access to these functionalities 🤓

**Review by KarolCislowski — APPROVED** [review]
Great API! 🦄 but please look into your frontend code because sending a message does not trigger useEffect() to update the list of messages.

**Comment by annalenksjo** [comment]
Thanks Karol! Any suggestions on how to fix that issue? I took a look at it
with my group but we dont know why the function is not running..

Den mån 31 maj 2021 kl 04:27 skrev Karol Cislowski ***@***.***

**Pull request closed without merging**